### ISSUE TITLE
Wrong point calculation and payment

### ISSUE THREAD

**Issue by paulodamaso** [zerocracy/farm#1609]
`0crat` made a wrong payment and point calculation in https://github.com/zerocracy/farm/pull/1601:
- Budget was 15 minutes (`REV`)
- Rate is $16 / hour
- Quality review was `good`
`0crat` paid me 25 points and $7.09 instead of 20 points and $5.32.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by g4s8** [comment]
@paulodamaso you've received 5 minutes extra for fast delivery as speed bonus (points and cash), and 5 minutes points for good quality + 33% cash. But you're right. It was calculated wrong, it was `(rate * minutes + speed_bonus) * 33%`, but should be `(rate * minutes) * 33% + speed_bonus`. It should be `$6.5` in your case, not `$7.09`.

**Comment by 0crat** [comment]
Job [#1609](https://github.com/zerocracy/farm/issues/1609) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
The job [#1609](https://github.com/zerocracy/farm/issues/1609) assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2291517694); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by paulodamaso** [comment]
@g4s8 I'd like to suggest to update policy with this explanation, after the 5 min - 33% change for good quality is not clear how much extra we receive in speed bonus.

**Comment by paulodamaso** [comment]
@g4s8 `(16 / 60 * 15) * 133 / 100 + (16 / 60 * 5) = 6.62` and not `6.5`, am I doing some wrong calculation here?

**Comment by g4s8** [comment]
@paulodamaso right, it should be `6.65`

**Comment by g4s8** [comment]
@paulodamaso I think it's too small (<$1) so you can ignore it.

**Comment by krzyk** [comment]
@g4s8 I think the % bonus messes up calculations too much. Couldn't we go back to straight +5min for good quality?

**Comment by krzyk** [comment]
@paulodamaso BTW. Speed bonus is described in policy at http://www.zerocracy.com/policy.html#36

**Comment by g4s8** [comment]
@krzyk +5min bonus payments has corner cases, e.g. if a user started a job, then he/she was kicked out from the project, then he completed the job with good quality, in such case we can't calculate bonus value because the user doesn't have role/rate in this project.

**Comment by krzyk** [comment]
@g4s8 but in such case +33% will also give nothing, because the user doesn't have a rate in project.

**Comment by g4s8** [comment]
@krzyk no so, 0crat calculates +33% from job estimate, not user's rate

**Comment by g4s8** [comment]
@krzyk @paulodamaso maybe it's better to calculate all bonuses on order start and add them to `estimates.xml`? In such way we can calculate bonus amount before task begin and avoid corner cases when user left project (https://github.com/zerocracy/farm/issues/1609#issuecomment-414076453). Also it looks more correct to lock project's funds for bonuses on order start, so project will be able to pay them even if it run out of funds.
@yegor256 what do you think about it?

**Comment by krzyk** [comment]
@g4s8 makes sense for me

**Comment by paulodamaso** [comment]
@g4s8 I agree too.

**Comment by g4s8** [comment]
@yegor256 ping, please see https://github.com/zerocracy/farm/issues/1609#issuecomment-414108799

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@0crat wait for clarification

**Comment by 0crat** [comment]
> @0crat wait for clarification  ([here](https://github.com/zerocracy/farm/issues/1609#issuecomment-415581451))

@carlosmiranda The impediment for [#1609](https://github.com/zerocracy/farm/issues/1609) was registered successfully by @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)

**Comment by carlosmiranda** [comment]
@yegor256 any word on this (https://github.com/zerocracy/farm/issues/1609#issuecomment-414108799)

**Comment by yegor256** [comment]
@g4s8 @carlosmiranda I agree with https://github.com/zerocracy/farm/issues/1609#issuecomment-414108799

**Comment by carlosmiranda** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/zerocracy/farm/issues/1609#issuecomment-434584099))

@carlosmiranda The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#1609](https://github.com/zerocracy/farm/issues/1609), please stop working. Reason for job resignation: Order was cancelled
